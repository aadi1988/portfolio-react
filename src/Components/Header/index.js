import React from 'react'
import Nav1 from '../Nav1'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faLaptop } from '@fortawesome/free-solid-svg-icons' 
function Header(props){
  const {
    categories = [],
    setCurrentCategory,
    currentCategory,
    portfolioSelected,
    setPortfolioSelected
  } = props;
     const laptop = <FontAwesomeIcon icon={faLaptop} size="1.5x"/>
     return(
        <div>
        <header>
          <h1 style={{color: "white"}}>{laptop}Aaditi Pai</h1>   
       
          <Nav1 
          categories={categories}
          setCurrentCategory={setCurrentCategory}
          currentCategory={currentCategory}
          portfolioSelected={portfolioSelected}
          setPortfolioSelected={setPortfolioSelected}
          />
        </header>
      </div>   
     );
}

export default Header;