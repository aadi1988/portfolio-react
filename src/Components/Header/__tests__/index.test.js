import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Header from '..'


afterEach(cleanup);
describe('Header component', () => {
    // baseline test
    it('renders', () => {
        render(<Header
          />);
    });
    // snapshot test
    it('matches snapshot', () => {
        const { asFragment } = render(<Header
            
          />);
        // assert value comparison
        expect(asFragment()).toMatchSnapshot();
    });
})