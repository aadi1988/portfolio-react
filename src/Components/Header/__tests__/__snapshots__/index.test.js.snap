// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <header>
      <h1
        style="color: white;"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-laptop fa-w-20 fa-1x "
          data-icon="laptop"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M624 416H381.54c-.74 19.81-14.71 32-32.74 32H288c-18.69 0-33.02-17.47-32.77-32H16c-8.8 0-16 7.2-16 16v16c0 35.2 28.8 64 64 64h512c35.2 0 64-28.8 64-64v-16c0-8.8-7.2-16-16-16zM576 48c0-26.4-21.6-48-48-48H112C85.6 0 64 21.6 64 48v336h512V48zm-64 272H128V64h384v256z"
            fill="currentColor"
          />
        </svg>
        Aaditi Pai
      </h1>
      <nav
        class="navbar navbar-expand-lg navbar-dark"
      >
        <button
          aria-controls="responsive-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="responsive-navbar-nav"
        >
          <div
            class="mr-auto navbar-nav"
          >
            <a
              class="mx-2 nav-link"
              href="#"
              role="button"
            />
            <a
              data-testid="cover"
              href="#"
            >
              Cover
            </a>
          </div>
        </div>
      </nav>
    </header>
  </div>
</DocumentFragment>
`;
